User dashboard #52

dietrich14 opened the pull request:
first PR
resolves #52 

Notes: 
- temporary endpoints for all tabs (myProcesses = all processes in the store - should be only the ones from the organization; processes in approval = all processes which are not yet approved - should be only the ones from the current user; approved processes = all processes which are already approved - should be only the ones from the current user)
- temporary processes in the news section as well

Commit:
Changed/Added following things:
- Changed entity names of *Impl.java files from uppercase to lowercase
- UserImpl changed, where email is now unique
- RegistrationServiceDatabaseImpl changed where email is set as the systemId and special symbols are replaced with an underline
- RegistrationServiceDatabaseImpl changed where the default role of a newly registered user is set to 'USER'

Commit:
added annotation for createdAt property in UserImpl to add automatic timestamp

Commit:
Added/Changed following things:
- added new csv's for database initialization (when using 'database' in application.properties) to store one admin user and all roles and rules
- RBACConfig is now ALWAYS running the RBACRetrievalService (as both database and memory strategy are performing an initial db insertion from csvs)
- RBACMappingService is now checking the property in the applications.properties file and depending on provided value, it gives the right path to the csvs (memoryusers or database_init)
- RBACRetrievalService is now getting the csvPath from the mappingService and is loading all rules,roles and users
- RBACRetrievalServiceMemoryImpl renamed to RBACRetrievalImpl as it is used for both strategies now

Commit:
added changes and additions in CHANGELOG.md

Commit:
Added/Changed:
- changed some property namings which caused false naming convention in json
- added annotations for createdAt to include in json and annot. to exclude irrelevant properties like password and systemid
- checked endpoints of UserController for user details /api/me and /api/user/userid => both working now as expected

Commit:
added/changed
- added OrganizationController
- added OrganizationService and Impl
- added OrganizationRepository and Impl
- changed RBACRepository where the organization related things have been removed
- changed mappings and annotations in the UserImpl and OrganizationImpl entity classes

Commit:
added/changed
- added endpoints for organization (getbyid, getall, save, update)
- added enpoints for user (update)
- changed mistake in OrganizationBuilder
- tested endpoints => working so far
- needs documentation in Swagger-API

Commit:
added jsonpropertyignore where employees are not returned in the organization property of a user to avoid infinite recursion; Updated CHANGELOG

Commit:
added pr url to changelog.md

Commit:
changed enpoints for login, register and checkIfMailTaken back to how it should be as the prefix /api is protected with token auth.

Commit:
Merge branch 'dev' into rbac-adaptations-#83

Commit:
=added tabs for my processes and processes in review

stfnltnr committed:
added endpoints to process-store, for detailed information check API documentation and pull requests

Changes to be committed:
        modified:   ../Gateway/src/main/java/at/fhjoanneum/ippr/gateway/api/controller/ProcessStoreGatewayController.java
        modified:   ../Gateway/src/main/java/at/fhjoanneum/ippr/gateway/api/controller/user/HttpHeaderUser.java
	modified:   ../Gateway/src/main/java/at/fhjoanneum/ippr/gateway/api/services/impl/ProcessStoreCallerImpl.java
	modified:   src/main/java/at/fhjoanneum/ippr/processstore/controller/ProcessStoreController.java
	modified:   src/main/java/at/fhjoanneum/ippr/processstore/persistence/entities/ProcessStoreBuilder.java
	modified:   src/main/java/at/fhjoanneum/ippr/processstore/persistence/entities/ProcessStoreObjectImpl.java
	modified:   src/main/java/at/fhjoanneum/ippr/processstore/repositories/ProcessStore.java
	modified:   src/main/java/at/fhjoanneum/ippr/processstore/services/ProcessStoreService.java
	modified:   src/main/java/at/fhjoanneum/ippr/processstore/services/ProcessStoreServiceImpl.java
	modified:   ../SpringCommons/src/main/java/at/fhjoanneum/ippr/commons/dto/processstore/ProcessStoreDTO.java

Commit:
=minor changes

stfnltnr committed:
gradle.bat and gradle-wrapper.properties for W10 startup

Commit:
=merged dev

Commit:
Merge branch 'dev' into store-service-#30

Commit:
=added validatedProcesses, design changes, added modal for process details

Commit:
=added modal for organization creation

Commit:
=added endpoint for saving organizations

Commit:
Merge branch 'dev' into rbac-adaptations-#83

Commit:
=functionality for creating new organizations; changed view rights for different users on dashboard

Commit:
Added:
- logic where the user who created an organization now gets the Role "ORG_CEO" assigned
- logic where a user who is being updated and where he joins a new org. gets the Role "ORG_EMP" assigned. This only works one time. Opting-out and changing org. does not work

Commit:
Merge branch 'rbac-adaptations-#83' of github.com:amarbajric/EBUSA-AIM17 into rbac-adaptations-#83

Commit:
=added toast after successfully saving the organization

Commit:
=bugfixes

Commit:
Merge branch 'dev' into user-dashboard-#52

Commit:
=edited changelod

Commit:
=linting

Commit:
=linting again

Commit:
=linting part 3

amarbajric commented:
**Review**
- Mixing german and english is not very intuitive and could lead to confusion.
  - Please change the language of the UI for your component to english
- Should there be an `Create Organization` button somewhere or will this not be part of the dashboard component?

Other than that, it looks promising to me and can be approved as soon as the points mentioned above are implemented/discussed.

amarbajric reviewed (changes requested):
See comment above

tortmann commented:
I definitely agree with the @amarbajric point to not mix DE and EN i forgot to mention that during our call today when we discussed the wording of the tabs to need changing

Commit:
=bugfixes

stfnltnr committed:
added Rating and ApprovalMessage

stfnltnr committed:
Merge branch 'dev' into store-service-#30

Commit:
Merge branch 'store-service-#30' of github.com:amarbajric/EBUSA-AIM17 into store-service-#30

Commit:
=changed language of the components to English

Commit:
=test refresh page

Commit:
=resolved conflicts

Commit:
Added endpoint where a file can be uploaded to a process

Commit:
=reload fail

Commit:
Added endpoint to download processFile

Commit:
added missing files

Commit:
Merge branch 'dev' into store-service-#30

Commit:
added changelog entry

stfnltnr committed:
added endpoints for mapping processes to orgaIds (BUYING)

Commit:
removed DELETE_FINAL folder

stfnltnr committed:
Merge branch 'store-service-#30' of github.com:amarbajric/EBUSA-AIM17 into store-service-#30

stfnltnr committed:
fixed issue with approved: null / false
Changes to be committed:
	modified:   ProcessStore/src/main/java/at/fhjoanneum/ippr/processstore/persistence/entities/ProcessStoreBuilder.java
	modified:   ProcessStore/src/main/java/at/fhjoanneum/ippr/processstore/persistence/entities/ProcessStoreObjectImpl.java
	modified:   ProcessStore/src/main/java/at/fhjoanneum/ippr/processstore/repositories/ProcessStore.java
        modified:   ProcessStore/src/main/resources/application.properties
	modified:   SpringCommons/src/main/java/at/fhjoanneum/ippr/commons/dto/processstore/ProcessStoreDTO.java

Commit:
Added properties for max file upload size which is now set to 10MB.

Commit:
Merge branch 'store-service-#30' of github.com:amarbajric/EBUSA-AIM17 into store-service-#30

Commit:
Merge branch 'dev' into store-service-#30

stfnltnr committed:
:poo: fuck gateway /api/store/process/create doenst send any data to the
store service

stfnltnr committed:
Merge branch 'store-service-#30' of github.com:amarbajric/EBUSA-AIM17 into store-service-#30

Commit:
added avgratingandcount endpoint for ratings

Commit:
Merge branch 'store-service-#30' of github.com:amarbajric/EBUSA-AIM17 into store-service-#30

Commit:
added controller of process store

Commit:
=resolved conflicts

Commit:
=added my processes (byOrga)

Commit:
=merge conflicts

Commit:
=started changing valid processes

Commit:
=fixed news

Commit:
fixing processstoredto

Commit:
Merge branch 'dev' of github.com:amarbajric/EBUSA-AIM17 into dev

Commit:
=merge conflicts

Commit:
=fixed in approval and not in approval

Commit:
=linting

Commit:
=linting

Commit:
Merge branch 'dev' into user-dashboard-#52

Commit:
changed lang from german to english
